src/init.c

```diff
@@ -357,39 +350,21 @@ int hyper_enter_sandbox(struct hyper_pod *pod, int pidpipe)
 		goto out;
 	}
 
-	if (setns(pidns, CLONE_NEWPID) < 0 ||
-	    setns(utsns, CLONE_NEWUTS) < 0 ||
+	if (setns(utsns, CLONE_NEWUTS) < 0 ||
 	    setns(ipcns, CLONE_NEWIPC) < 0) {
 		perror("fail to enter the sandbox");
 		goto out;
 	}
 
-	/* current process isn't in the pidns even setns(pidns, CLONE_NEWPID)
-	 * was called. fork() is needed, so that the child process will run in
-	 * the pidns, see man 2 setns */
-	ret = fork();
-	if (ret < 0) {
-		perror("fail to fork");
-		goto out;
-	} else if (ret > 0) {
-		fprintf(stdout, "create child process pid=%d in the sandbox\n", ret);
-		if (pidpipe > 0) {
-			hyper_send_type(pidpipe, ret);
-		}
-		_exit(0);
-	}
 
 out:
-	if (pidns >= 0)
-		close(pidns);
-
 	if (ipcns >= 0)
 		close(ipcns);
 
 	if (utsns >= 0)
 		close(utsns);
 
-	return ret;
+	return 0;
```

Comment:
This should be kept as return ret, as we could reach here from failing to open the uts/ipc paths or from setns failures.
Should set ret=0 before the out goto label.